pkg/scheduler/objects/application.go

```diff
@@ -281,6 +303,9 @@ func (sa *Application) timeoutStateTimer(expectedState string, event application
 				sa.notifyRMAllocationReleased(sa.rmID, toRelease, si.TerminationType_TIMEOUT, "releasing placeholders on app complete")
 				sa.clearStateTimer()
 			} else {
+				// HandleApplicationEvent assumes the lock to be held
+				sa.Lock()
+				defer sa.Unlock()
```

Comment:
We have two choices: 
1. move adding this lock and a lock review of `timeoutStateTimer()` to a new jira., or
2. fix all locking issues here: `getPlaceholderAllocations()` accesses the allocations map unlocked which might cause issues. `clearStateTimer()` in all cases but here is called under lock.
We discussed the `clearStateTimer()` call yesterday and I am not sure it causes a problem but we might as well be careful.